### PR TITLE
Add validation for new asset proposal validation timestamp

### LINKED ISSUE #5818: A New Asset proposal submitted without a Validation Timestamp should be rejected
See https://github.com/vegaprotocol/vega/issues/5816 for some background

# To recreate
```bash
./vegawallet command send --wallet test --pubkey 6eed3515524b3c6dda7a9eceaebba58e236a56faa8ef7b8129179c9d269f41f2 --network fairground '{
   "proposalSubmission": {
    "rationale": {
     "description": "Add DAI which definitely exists"
    },
    "terms": {
     "newAsset": {
      "changes": {
       "name": "Dai Stablecoin",
       "symbol": "DAI",
       "totalSupply": "49819719369171782033318831",
       "decimals": "18",
       "quantum": "1",
       "erc20": {
        "contractAddress": "0x31f42841c2db5173425b5223809cf3a38fede360",
        "withdrawThreshold": "10",
        "lifetimeLimit": "10"
       }
      }
     },
     "closingTimestamp": 1659978907,
     "enactmentTimestamp": 1659979607
    }
   }
  }'
```

# Expected output
Transaction is rejected as it is missing a field

# Actual result
Transaction is accepted, validationTimestamp is set to 0. What does that mean? :woman_shrugging: https://github.com/vegaprotocol/vega/blob/0b92db2ccd195a29731b01e7d8801cca6c06edf6/protos/vega/governance.pb.go#L1370-L1371 :man_shrugging:. But it shouldn't be 0.

# Tasks
- [ ] Reject New Asset transactions (maybe all, maybe just ERC20 - I dunno) that don't have a validationTimestamp set

### LINKED ISSUE #5819: [wallet] Wallet validation should catch that New Asset proposals for ERC20 assets require a validationTimestamp
See #5818 

# Tasks
- [ ] After #5818 ensure that a New Asset transaction for an ERC20 asset that does not have a `validationTimestamp` is rejected

### PR DESCRIPTION
**Pull request by jeremyletang**
close #5819 
close #5818

### CODE REVIEW

**Comment by vega-paul** [comment]
`{"vega":"bugfix/5819-5818"}

**Comment by vega-paul** [comment]
full run https://jenkins.ops.vega.xyz/job/common/job/system-tests/7306/

File: commands/proposal_submission.go [hunk 1]
```diff
@@ -96,6 +97,23 @@ func checkProposalSubmission(cmd *commandspb.ProposalSubmission) Errors {
...
+		}
+		if cmd.Terms.ValidationTimestamp > cmd.Terms.ClosingTimestamp {
+			errs.AddForProperty("proposal_submission.terms.validation_timestamp",
+				errors.New("cannot be after closing time"),
+			)
+		}
+	default:
+		if cmd.Terms.ValidationTimestamp != 0 {
+			errs.AddForProperty("proposal_submission.terms.validation_timestamp", ErrIsNotValid)
+		}
```
Comment (ValentinTrinque) [inline commands/proposal_submission.go:114]: I think it should be something like "Is not supported" instead.

File: commands/proposal_submission_new_asset_test.go [hunk 2]
```diff
@@ -33,6 +34,45 @@ func TestCheckProposalSubmissionForNewAsset(t *testing.T) {
 	t.Run("Submitting an ERC20 asset change with valid lifetime limit succeeds", testNewERC20AssetChangeSubmissionWithValidLifetimeLimitSucceeds)
 	t.Run("Submitting an ERC20 asset change with invalid withdrawal threshold fails", testNewERC20AssetChangeSubmissionWithInvalidWithdrawalThresholdFails)
 	t.Run("Submitting an ERC20 asset change with valid withdrawal threshold succeeds", testNewERC20AssetChangeSubmissionWithValidWithdrawalThresholdSucceeds)
+	t.Run("Submitting an ERC20 asset change without validation timestamp fails", testNewAssetERC20ChangeSubmissionMissingValidationTimestamp)
+	t.Run("Submitting an ERC20 asset change with validation timestamp succeed", testNewAssetERC20ChangeSubmissionWithValidationTimestampSucceeds)
+	t.Run("Submitting an ERC20 asset change with validation after closing timestamp fails", testNewAssetERC20ChangeSubmissionValidationAfterClosingTimestampsFails)
```
Comment (ValentinTrinque) [inline commands/proposal_submission_new_asset_test.go:39]: We are missing the default branch where Validation timestamp shouldn't be set.

**Review by ValentinTrinque — CHANGES_REQUESTED** [review]

**Review by ze97286 — DISMISSED** [review]

File: commands/proposal_submission_new_asset_test.go (hunk 2 above)
Comment (jeremyletang) [reply commands/proposal_submission_new_asset_test.go:39]: right will add one

**Review by ValentinTrinque — APPROVED** [review]

**Pull request merged**